### DISCUSSION TITLE
Numerical instability error occurred during methane jet fire model

### DISCUSSION THREAD

**Discussion by DOGzym** [firemodels/fds#12674] [Q&A]
Hello everyone,
A horizontal jet fire model with complete combustion of methane has been established. The speed of 21.2m/s, a nozzle radius of 2.5mm, and a grid size of 0.5mm are setting as the initial condition in this model. During the operation, 'WARNING: Minimum density, 0.116kg/m3, clipped in Mesh1' occurred. The program unexpectedly stopped running because of numerical instability.

I think the grid size is too small and the nozzle velocity is too high, resulting in the grid not being able to capture methane gas. So how to solve the problem of numerical instability while ensuring 0.5mm accuracy?

The code is as follows:

&HEAD CHID = 'JETFIRE19'
      TITLE = 'Horizonal jet fire'   /
        
&MESH ID = 'MESH01'
      XB = 0 0.0500 -0.0150 0.0150 0 0.0400
      IJK = 100 60 80 /

&MISC TMPA = 30 /
&DUMP DT_RESTART = 5 /
&TIME T_END = 30 /
&VENT MB = XMIN          SURF_ID = 'OPEN' /
&VENT MB = XMAX          SURF_ID = 'OPEN' /
&VENT MB = YMIN          SURF_ID = 'OPEN' /
&VENT MB = YMAX          SURF_ID = 'OPEN' /
&VENT MB = ZMIN          SURF_ID = 'OPEN' /
&VENT MB = ZMAX          SURF_ID = 'OPEN' /

&SPEC ID = 'METHANE'  /
&SPEC ID = 'OXYGEN'          LUMPED_COMPONENT_ONLY = F /
&SPEC ID = 'NITROGEN'        LUMPED_COMPONENT_ONLY = T / 
&SPEC ID = 'WATER VAPOR'     LUMPED_COMPONENT_ONLY = T /
&SPEC ID = 'CARBON DIOXIDE'  LUMPED_COMPONENT_ONLY = T / 
&SPEC ID = 'AIR'  BACKGROUND = T 
      SPEC_ID(1) = 'OXYGEN' 
      VOLUME_FRACTION(1) = 2
      SPEC_ID(2) = 'NITROGEN' 
      VOLUME_FRACTION(2) = 7.52  /
&SPEC ID = 'PRODUCTS' 
      SPEC_ID (1) = 'CARBON DIOXIDE' 
      VOLUME_FRACTION (1) = 1
      SPEC_ID (2) = 'WATER VAPOR'     
      VOLUME_FRACTION (2) = 2
      SPEC_ID (3) = 'NITROGEN'        
      VOLUME_FRACTION (3) = 7.52 /
&REAC FUEL = 'METHANE'
      HEAT_OF_COMBUSTION = 55644.375
      SPEC_ID_NU = 'METHANE','AIR','PRODUCTS'
      NU = -1,-1,1 /

&SURF ID = 'BURNER'
      SPEC_ID(1) = 'METHANE'   
      MASS_FRACTION(1) = 1
      VEL = -21.2
      RGB = 255 0 0 /
&OBST XB = 0 0.001 -0.0025 0.0025 0.0020 0.0070
      RGB = 128 64 64 
      SURF_ID = 'INERT' /    
&VENT ID = 'PLANE'
      XB = 0.001 0.001 -0.0025 0.0025 0.0020 0.0070
      XYZ = 0.001 0 0.0045
      RADIUS = 0.0025
      SURF_ID = 'BURNER' /

&SLCF PBY = 0
      QUANTITY = 'TEMPERATURE'
      VECTOR = T /
&SLCF PBY = 0
      QUANTITY = 'VELOCITY'
      VECTOR = T /
&SLCF PBX = 0.0020
      QUANTITY = 'TEMPERATURE'
      VECTOR = T /
&SLCF PBX = 0.0020
      QUANTITY = 'VELOCITY'
      VECTOR = T /
&SLCF PBX = 0.0025
      QUANTITY = 'TEMPERATURE'
      VECTOR = T /
&SLCF PBX = 0.0025
      QUANTITY = 'VELOCITY'
      VECTOR = T /
&SLCF PBX = 0.0030
      QUANTITY = 'TEMPERATURE'
      VECTOR = T /
&SLCF PBX = 0.0030
      QUANTITY = 'VELOCITY'
      VECTOR = T /
      
&DEVC ID = 'TIMER'     XYZ = 0.0020 0.0005 0.0020     SETPOINT = 0.5   QUANTITY = 'TIME'   INITIAL_STATE = T /
&SURF ID = 'IGNITER'
      TMP_FRONT = 2500
      COLOR = 'GREEN' /
&OBST XB = 0.0015 0.0020 0 0.0005 0.0020 0.0025
      DEVC_ID = 'TIMER'
      SURF_ID6 = 'IGNITER','IGNITER','IGNITER','IGNITER','IGNITER','IGNITER' /

**Comment by drjfloyd** [comment]
What version of FDS are you using? We just released FDS 6.9.0.

With 0.5 mm grid cells you should add SIMULATION_MODE='DNS' to the MISC line.

A couple of other observations (probably not part of your issue), you don't need to define the PRODUCTS and AIR species. You could just use:

&REAC FUEL='METHANE'/

FDS at that point will know what METHANE is since it is a predefined species and will know it is CH4 and FDS will already define AIR as a default.  Not that it is incorrect to do it how you did it, just that it is more work for you than is absolutely needed when using a predefined fuel that just contains C,H,O, and N.

Your HEAT_OF_COMBUSTION for METHANE is too high.  You are using the higher heating value rather than lower heating value of ~50 MJ/kg (which is the appropriate value for FDS).

**Comment by drjfloyd** [comment]
You also don't need an ignition source since you are using fast chemistry without setting an AUTO_IGNITION_TEMPERATURE.
